crates/router/src/connector/braintree/transformers.rs

```diff
@@ -255,234 +544,1268 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+#[derive(Default, Debug, Clone, Serialize)]
+pub struct BraintreeCaptureRequest {
+    query: String,
+    variables: VariableCaptureInput,
+}
+
+impl TryFrom<&BraintreeRouterData<&types::PaymentsCaptureRouterData>> for BraintreeCaptureRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &BraintreeRouterData<&types::PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let query = CAPTURE_TRANSACTION_MUTATION.to_string();
+        let variables = VariableCaptureInput {
+            input: CaptureInputData {
+                transaction_id: item.router_data.request.connector_transaction_id.clone(),
+                transaction: CaptureTransactionBody {
+                    amount: item.amount.to_owned(),
+                },
+            },
+        };
+        Ok(Self { query, variables })
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CaptureResponseTransactionBody {
+    id: String,
+    status: BraintreePaymentStatus,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CaptureTransactionData {
+    transaction: CaptureResponseTransactionBody,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CaptureResponseData {
+    capture_transaction: CaptureTransactionData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CaptureResponse {
+    data: CaptureResponseData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum BraintreeCaptureResponse {
+    CaptureResponse(Box<CaptureResponse>),
+    ErrorResponse(Box<ErrorResponse>),
+}
+
+impl TryFrom<types::PaymentsCaptureResponseRouterData<BraintreeCaptureResponse>>
+    for types::PaymentsCaptureRouterData
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::PaymentsCaptureResponseRouterData<BraintreeCaptureResponse>,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            BraintreeCaptureResponse::CaptureResponse(capture_data) => {
+                let transaction_data = capture_data.data.capture_transaction.transaction;
+                let status = enums::AttemptStatus::from(transaction_data.status.clone());
+                let response = if utils::is_payment_failure(status) {
+                    Err(types::ErrorResponse {
+                        code: transaction_data.status.to_string().clone(),
+                        message: transaction_data.status.to_string().clone(),
+                        reason: None,
+                        attempt_status: None,
+                        connector_transaction_id: Some(transaction_data.id),
+                        status_code: item.http_code,
+                    })
+                } else {
+                    Ok(types::PaymentsResponseData::TransactionResponse {
+                        resource_id: types::ResponseId::ConnectorTransactionId(transaction_data.id),
```

Comment:
Do we need to update the transaction id after capture call?

Reply:
Not needed, Its returning the same transaction id

Reply:
Remove this in subsequent PRs